### ISSUE TITLE
String input doing weird things with {}

### ISSUE THREAD

**Issue by shadownetdev1** [comfyanonymous/ComfyUI#679]
given the input string "{apple|pizza}"
```
return {
    "required": {
        "text": ("STRING", {"default": "",})
    }
}
```
will return: "{apple|pizza}"
```
return {
    "required": {
        "text": ("STRING", {"default": "", "multiline": True,})
    }
}
```
will return "apple" or "pizza"

What I would expect to happen is for both to return "{apple|pizza}" with a potential optional bool argument "parse_dynamic_prompt" (or something) for when we want to do the parsing that "multiline" = True is currently doing. I wouldn't actually have any problems with multiline text input parsing this if it did it the same as dynamic prompts, but it doesn't. Given the input "{1-2$$apple|pizza}" multiline returns "1-2$$apple" or "pizza". If someone would point me to the relevant lines I would be willing to fix it and create a pull request.

**Comment by shadownetdev1** [comment]
My test code in it's entirety:
```
class MultilineStringNode:
    def __init__(self):
        pass

    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "Text": ("STRING", {
                    "default": "",
                    "multiline": True,
                }),
            }
        }

    RETURN_TYPES = ("STRING",)
    FUNCTION = "get_value"
    CATEGORY = "SND1/test"

    def get_value(self, Text):
        return (Text,)


class SinglelineStringNode:
    def __init__(self):
        pass

    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "Text": ("STRING", {
                    "default": "",
                    "multiline": False,
                }),
            }
        }

    RETURN_TYPES = ("STRING",)
    FUNCTION = "get_value"
    CATEGORY = "SND1/test"

    def get_value(self, Text):
        return (Text,)


NODE_CLASS_MAPPINGS = {
    "SND1_MultilineStringNode": MultilineStringNode,
    "SND1_SinglelineStringNode": SinglelineStringNode,
}
```

**Comment by randomeizer** [comment]
Yeah, I've encountered this same issue. It seems that the issue is that all multiline text fields are being treated as "dynamic prompts", where it will do some random selection of "{ foo | bar }" values within curly brackets. This is done in the `web/extensions/core/dynamicPrompts.js` file.

  Currently, it will default to "true" for multiline (aka "customtext") fields, and false for anything else. It requires some Javascript code changes in the `addMultilineWidget()` function in `web/scripts/widgets.js` to allow the Python node definition to actually make "dynamicPrompts" a configurable option for multiline text:

```javascript
widget.dynamicPrompts = opts.dynamicPrompts;
```

Then, your node definition can pass in `dynamicPrompts` as an option on the input field:

```python
class RBTextDebug:
    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "label": ("STRING", {"default": "Text Value"}),
                "text": ("STRING", {"multiline": True, "dynamicPrompts": False}),
            }
        }

    RETURN_TYPES = ("STRING",)
    OUTPUT_NODE = True
    FUNCTION = "debug"
    CATEGORY = "Example/Debug"

    def debug(self, label, text):
        print(f"{label}: {text}")
        return (text, )
```

All that said, I would strongly vote for 'dynamicPrompts' to default to `False` for everything, and to only opt-in. That would require updating all the standard prompt nodes to support it, and any custom libraries would also have to update. Feels like the correct thing to do though.

**Comment by freem-trg** [comment]
Hello!
I also encountered a similar problem when I tried to build a custom pipe using BLIP, and for a long time I could not understand why the node did not work as intended. The only way was to add print() in different places :)... I don't have much experience in writing custom nodes, is it possible to make a debug node that will receive text at the input and show the received value in some field inside itself? It seems to me that there is a lack of a standard node that could simply display some intermediate prompt data.

**Comment by randomeizer** [comment]
I've created a pull request here at #1042 to allow opting out of "dynamic prompts" behaviour.

As noted, I'd prefer to make it opt in.

**Comment by TMRolle** [comment]
I'm running into this as well while trying to implement custom nodes for Langchain integration. Langchain uses python f-string format for prompt template variables so it's a pretty major blocker :/

**Comment by randomeizer** [comment]
Yeah, this was pretty much what tripped me up too. Not for Langchain, but using f-string format.

Would be great if a fix can get checked in.

**Comment by doctorpangloss** [comment]
This should be closed, because dynamicPrompts can be set, but it should be fixed to be default False, not default True
